components/tidb_query_datatype/src/codec/mysql/json/json_contains.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Ordering;
+
+use super::{super::Result, JsonRef, JsonType};
+
+impl<'a> JsonRef<'a> {
+    /// `json_contains` is the implementation for JSON_CONTAINS in mysql
+    /// <https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-contains>
+    /// See `ContainsBinary()` in TiDB `json/binary_function.go`
+    pub fn json_contains(&self, target: JsonRef<'_>) -> Result<Option<bool>> {
```

Comment:
Why is it a `Result<Option<bool>>`? It seems that it will only return `Ok(Some(true))`, `Ok(None)` and `Err()`. Could it be a `Result<bool>`?

Reply:
I changed Result<Option<bool>> to Result<bool>